packages/custom_lint/lib/src/package_utils.dart

```diff
@@ -115,14 +123,75 @@ Future<PackageConfig?> tryParsePackageConfig(Directory directory) async {
   }
 }
 
+/// Finds the project directory associated with an analysis context root
+///
+/// This is a folder that contains both a `pubspec.yaml` and a `.dart_tool/package_config.json` file.
+/// It is either alongside the analysis_options.yaml file, or in a parent directory.
+Directory findProjectDirectory(
```

Comment:
This handles looking for a project directory which contains at minimum a pubspec next to a .dart_tool/package_config.json (and in the case of hosted dependencies in the pubcache, the package config file is not necessary).

Reply:
Why do we care about searching for the existence of a `package_config.json` here? Shouldn't searching only for a `pubspec` be enough?

Reply:
I'm not sure why this was resolved. I still don't get why we are looking for package_config.jsons 

To begin with, there's a high chance that if there's a missing package_config, then https://github.com/invertase/dart_custom_lint/pull/168 should take care of it.